google/cloud/aiplatform/models.py

```diff
@@ -215,10 +229,525 @@ def upload(
+                percentage of this Endpoint's traffic that
+                should be forwarded to that DeployedModel.
+                If a DeployedModel's ID is not listed in this
+                map, then it receives no traffic.
+
+                The traffic percentage values must add up to
+                100, or map must be empty if the Endpoint is to
+                not accept any traffic at a moment.
+            labels (Dict):
+                Optional. The labels with user-defined metadata to
+                organize your Endpoints.
+                Label keys and values can be no longer than 64
+                characters (Unicode codepoints), can only
+                contain lowercase letters, numeric characters,
+                underscores and dashes. International characters
+                are allowed.
+                See https://goo.gl/xmQnxf for more information
+                and examples of labels.
+            metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as
+                metadata.
+            project (str):
+                Optional. Project to retrieve endpoint from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to retrieve endpoint from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        Returns:
+            endpoint (Endpoint):
+                Instantiated representation of the endpoint resource.
+        """
+
+        endpoint_client = cls._instantiate_client(
+            location=location, credentials=credentials
+        )
+
+        client_class = PredictionServiceClient
+        _is_client_prediction_client = True
+        prediction_client = cls._instantiate_client(
+            location=location, credentials=credentials
+        )
+
+        create_endpoint_operation = cls._create(
+            endpoint_client=endpoint_client,
+            parent=initializer.global_config.common_location_path(
+                project=project, location=location
+            ),
+            display_name=display_name,
+            description=description,
+            traffic_split=traffic_split,
+            labels=labels,
+            metadata=metadata,
+        )
+
+        created_endpoint = None
+        endpoint = cls(
+            endpoint=created_endpoint,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        create_endpoint_operation.add_update_resource_callback(
+            resource_noun_obj=endpoint,
+            result_key="name",
+            api_get=lambda name: endpoint_client.get_endpoint(name=name),
+        )
+
```

Comment:
Preference to block here and add async support later.

Reply:
makes sense and will be cleaner for operations that require endpoint.